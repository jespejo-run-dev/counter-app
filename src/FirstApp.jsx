import PropTypes from 'prop-types';

export const FristApp = ({
    name = 'Jespejo',
    subTitle = 'no hay subtítulo',
    title = 'No hay título',
}) => {

    // console.log(props);

    return (
        <>
            <div data-testid="test-title">{title}</div>
            {/* Expresion no objeto */}
            {/* <code>{JSON.stringify(newMessage)}</code> */}
            <p>{subTitle}</p>
            <p>{subTitle}</p>
            <p>{name}</p>
        </>
    )
}

FristApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string,
}
