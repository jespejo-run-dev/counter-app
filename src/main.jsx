import React from 'react'
import ReactDOM from'react-dom/client'
import { FristApp } from './FirstApp'
import { CounterApp } from './CounterApp'
import './styles.css'


ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <FristApp title='Hola, Jespejo' />
        {/* <CounterApp value={5} /> */}
    </React.StrictMode>
)