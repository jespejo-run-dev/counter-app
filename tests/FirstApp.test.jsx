import { render } from "@testing-library/react";
import { FristApp } from "../src/FirstApp";

describe('Pruebas en <FirstApp />', () => {
  
    test('Debe hacer match con el snapshot', () => {
        
        const title="Hola, Jespejo";
        const {container} = render(<FristApp title={title} />)

        expect(container).toMatchSnapshot();
        
        
    });
    
    test('debe mostrar el titulo en un h1', () => {
        const title="Hola, Jespejo";
        const {container, getByText} = render(<FristApp title={title} />)

        expect(getByText(title)).toBeTruthy();

        const h1 = container.querySelector('h1');
        expect(h1.innerHTML).toContain(title);
        

    })
    

});
