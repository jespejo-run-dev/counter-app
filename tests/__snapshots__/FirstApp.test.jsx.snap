// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola, Jespejo
  </h1>
  <p>
    no hay subt√≠tulo
  </p>
  <p>
    Jespejo
  </p>
</div>
`;
